### ISSUE TITLE
Change Request: Replace `graphemer` with `Intl.Segmenter`

### ISSUE THREAD

**Issue by ota-meshi** [eslint-stylistic/eslint-stylistic#219]
### Clear and concise description of the problem

I also posted this issue on ESLint.
https://github.com/eslint/eslint/issues/17835

If we use the Node.js>=16 we can use `Intl.Segmenter` instead of `graphemer`.
`Intl.Segmenter` can be used to reduce the dependency packages of this plugin.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segmenter#browser_compatibility

### Suggested solution

Replace `graphemer` with `Intl.Segmenter`.

before:

```js
    if (!splitter) {
        splitter = new Graphemer();
    }

    return splitter.countGraphemes(value);
```

after:

```js
    if (!segmenter) {
        segmenter = new Intl.Segmenter();
    }

    return [...segmenter.segment(value)].length;
```

### Additional context

I tested the compatibility of `graphemer` and `Intl.Segmenter`. The test case in the `graphemer` repository did not change the result even if I replaced it with `Intl.Segmenter`.
However, "अनुच्छेद" in the README has a different result.

`graphemer`: `'अ', 'नु', 'च्', 'छे', 'द'`
`Intl.Segmenter`: `'अ', 'नु', 'च्छे', 'द'`

I think this difference is probably due to the difference in supported unicode versions.

`graphemer` supports unicode v15.0, but tests fail with v15.1.
https://github.com/eslint/eslint/issues/17835#issuecomment-1849748312

---

Repositories used for testing:
https://github.com/ota-meshi/graphemer/tree/compat-intl-segmenter
Commit:
https://github.com/ota-meshi/graphemer/commit/de301e6079e7e4d179a7ed88113e8e406033e77b

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [X] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by ota-meshi** [comment]
I'm not sure if this is considered a breaking change :thinking: What do you think?

**Comment by antfu** [comment]
We already on node 16, so I'd love to switch over if they behave the same 👍
https://github.com/eslint-stylistic/eslint-stylistic/blob/3229635dfdb4d2fd007fb11338ae98e488094c0f/packages/eslint-plugin-js/package.json#L103-L105

I'd consider this a performance improvement or a fix (to Unicode 15.1) rather than a breaking change.

**Comment by ota-meshi** [comment]
Thanks for your comment! I will submit a PR.
